mlflow/server/handlers.py

```diff
@@ -1119,6 +1124,48 @@ def search_datasets_handler():
         return _not_implemented()
 
 
+@catch_mlflow_exception
+def gateway_proxy_handler():
+    args = request.json
+    host = args.get("host")
+    if not host:
+        raise MlflowException(
+            message="GatewayProxy request must specify a host.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    port = args.get("port")
+    if not port:
+        raise MlflowException(
+            message="GatewayProxy request must specify a port.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    gateway_path = args.get("gateway_path")
+    if not gateway_path:
+        raise MlflowException(
+            message="GatewayProxy request must specify a gateway_path.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    request_type = args.get("request_type")
+    if not request_type:
+        raise MlflowException(
+            message="GatewayProxy request must specify a request_type.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
```

Comment:
Can this be inferred from the request itself? I.e. does Flask give us any utils to determine this? Would be nice if the caller doesn't have to send a separate request type field

Reply:
Good point. Yes, we can get this from `request.method`